crawlers/tokensScanner.ts

```diff
@@ -0,0 +1,143 @@
+import pino, { Logger } from 'pino';
+import { Sequelize } from 'sequelize/types';
+import { OpalAPI } from '../lib/providerAPI/bridgeProviderAPI/concreate/opalAPI';
+import { TestnetAPI } from '../lib/providerAPI/bridgeProviderAPI/concreate/testnetAPI';
+import { getFormattedToken } from '../lib/token/tokenData';
+import { getCollectionsSchemaInfo } from '../lib/collection/collectionDB';
+import { ITokenDB } from '../lib/token/tokenDB.interface';
+import { BridgeAPI } from '../lib/providerAPI/bridgeApi';
+import { save as saveTokenDb, del as delTokenDb } from '../lib/token/tokenDB';
+import { ICrawlerModuleConstructorArgs, ICollectionSchemaInfo } from './crawlers.interfaces';
+
+class TokensScanner {
+  private logger: Logger;
+
+  private sequelize: Sequelize;
+
+  private bridgeApi: OpalAPI | TestnetAPI;
+
+  constructor({ logger } : { logger: Logger }) {
+    this.logger = logger;
+  }
+
+  private async getCollectionTokens(collectionInfo: ICollectionSchemaInfo) {
+    const { collectionId } = collectionInfo;
+    // todo: debug
+    // const tokensCount = Math.min((await this.bridgeApi.getTokenCount(collectionId)), 2);
```

Comment:
Why 2 is minimum value? What if we haven't tokens in collection at all?

Reply:
Oh it's min, not max. Something strange happening in this place.

Reply:
It was not very smart debug take, but worked for me) Removed.